prow/plugins/tbr/tbr.go

```diff
@@ -0,0 +1,142 @@
+/*
+Copyright 2018 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package tbr
+
+import (
+	"fmt"
+	"regexp"
+
+	"github.com/sirupsen/logrus"
+
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+	"k8s.io/test-infra/prow/repoowners"
+)
+
+const PluginName = "tbr"
+
+var (
+	// TODO(bentheelder,cjwagner): move label definitions to a common package
+	LGTMLabel     = "lgtm"
```

Comment:
imo we should pull out all of these plugin labels into a common package so we can define them once. currently it's a bit awkward for plugins that interact with the same labels

Reply:
Thoughts on putting this logic into lgtm?

Reply:
Aka this seems to me like an extension of the lgtm logic, which can be enabled/disabled his config like approve = lgtm

Reply:
It'd need to be off by default imo, with per-repo config enabling it, I don't think larger projects need or should enable this functionality.

Also that plugin is hairier to work on, there's a lot of messy logic imo :/

Reply:
OK, I can do that. To me it's slightly different because it requires existing approval and only allows the author to do this (other users should just lgtm), whereas LGTM does not.

Reply:
This doesn't skip the approval requirement. You can only tbr code you own. 

This is just an emergency way to add an lgtm label. 

(I would expect tbr to add lgtm even if I don't have a approve label.  I'd also expect it not to merge until it has an approve label)

Reply:
Ok, re-reading this, so then we can do:

- works just like LGTM, but only for the PR author, only if enabled in plugins config, and adds another label marking it as tbr

to merge you still need approval through the normal mechanism

SGTM, that's cleaner than my first reading which somehow had us checking for approval permissions in the lgtm plugin for TBR.. 

PR inbound

Reply:
https://github.com/kubernetes/test-infra/pull/9424, needs test coverage